pkg/s3select/json-input.go

```diff
@@ -0,0 +1,143 @@
+/*
+ * Minio Cloud Storage, (C) 2018 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package s3select
+
+import (
+	"encoding/json"
+	"encoding/xml"
+	"io"
+
+	jsoniter "github.com/json-iterator/go"
+)
+
+// NewJSONInput sets up a new Input, the first Json is read when this is run.
+// If there is a problem with reading the first Json, the error is returned.
+// Otherwise, the returned reader can be reliably consumed with jsonRead()
+// until jsonRead() returns nil.
+func NewJSONInput(opts1 *JSONOptions) (*JSONInput, error) {
+
+	myReader := opts1.ReadFrom
+	var tempBytesScanned int64
+	progress := &statInfo{
+		BytesScanned:   tempBytesScanned,
```

Comment:
`BytesScanned:   tempBytesScanned,` can be changed to `BytesScanned:   int64(0),`

Reply:
Didn't modify

Reply:
I see that the code has changed and this comment is not relevant anymore.